src/main/java/org/candlepin/model/ContentOverrideCurator.java

```diff
@@ -56,35 +56,49 @@ public List<T> getList(Parent parent) {
     }
 
     public void removeByName(Parent parent, String contentLabel, String name) {
-        List<T> overrides = currentSession()
-            .createCriteria(this.entityType())
-            .add(Restrictions.eq(parentAttr, parent))
-            .add(Restrictions.eq("contentLabel", contentLabel))
-            .add(Restrictions.eq("name", name).ignoreCase())
-            .list();
+        CriteriaBuilder cb = getEntityManager().getCriteriaBuilder();
```

Comment:
While this works, we could instead just do a direct delete either with JPQL or a CriteriaDelete, rather than fetching everything and then deleting them one-by-one.

This occurs in a few places.

Reply:
done